homeassistant/components/nibe_heatpump/climate.py

```diff
@@ -0,0 +1,349 @@
+"""The Nibe Heat Pump climate."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from nibe.coil import Coil
+from nibe.exceptions import CoilNotFoundException
+
+from homeassistant.components.climate import (
+    ATTR_HVAC_MODE,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    ATTR_TEMPERATURE,
+    ClimateEntity,
+    ClimateEntityDescription,
+    ClimateEntityFeature,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import DOMAIN, LOGGER, Coordinator
+from .const import Series
+
+
+@dataclass
+class ClimateDescriptionMixin:
+    """Mixin for required fields."""
+
+    current_address: int
+    setpoint_heat_address: int
+    setpoint_cool_address: int
+    prio_address: int
+    mixing_valve_state_address: int
+    active_accessory_address: int | None
+    use_room_sensor_address: int
+    cooling_with_room_sensor_address: int
+
+
+@dataclass
+class ClimateDescription(ClimateEntityDescription, ClimateDescriptionMixin):
+    """Base description."""
+
+
+ADDRESS_PRIO_F = 43086
+ADDRESS_PRIO_S = 31029
+
+ADDRESS_COOLING_WITH_ROOM_SENSOR_F = 47340
+ADDRESS_COOLING_WITH_ROOM_SENSOR_S = 40171
+
```

Comment:
I'd move all the magic numbers to the lib.

Reply:
They are really just as magic in the library thou. My pump have about 900 parameters. Adding them all in library makes little sense. We do have string names for these like "prio-43086" and "cooling-with-room-sensors-40171" which we could use instead.

They are a little bit better, BUT they stem from a DB export from the pump that is language translated. So nibe could decide, oh we don't like that naming and we end up with new names. 

Also the names are not always consistent, which sort of make it more confusing. The number is consistent within a series of pumps.

I've been a bit back and forth on this. The library have no need to know these constants. Only a application that want to expose a climate entity in the format that home assistant exposes.

I will ponder it some more thou.

Reply:
I've moved them to lib